src/components/Draggable/RealFilterDraggable.tsx

```diff
@@ -0,0 +1,205 @@
+import React, { useState } from 'react'
+import { useDrag } from 'react-dnd'
+import { FixedSizeList } from 'react-window'
+import { Button, Dropdown, DropdownItem, HFlow, Icon, TextField } from '..'
+import { Theme, useStyles, useTheme } from '../../styles'
+import { FilterDraggableProps } from './FilterDraggable'
+import { QuantityEnum } from './types/QuantityEnum'
+import { ItemTypes } from './types/ItemTypes'
+import { getKeyDirection } from './util'
+import { DraggableRow } from './DraggableRow'
+
+export function RealFilterDraggable<T>(props: FilterDraggableProps<T>) {
+  const { name, origin, value, filterValues, filterState, onDragEnd, handleFilterUpdate, formatter, onKeyNav } = props
+
+  const [searchedFilterSet, setSearchedFilterSet] = useState<Array<string>>(filterValues)
+
+  const [open, setOpen] = useState(false)
+
+  const [all, setAll] = useState<QuantityEnum.EMPTY | QuantityEnum.HALF_FULL | QuantityEnum.FULL>(
+    filterState.size === 0
+      ? QuantityEnum.EMPTY
+      : filterState.size === filterValues.length
+      ? QuantityEnum.FULL
+      : QuantityEnum.HALF_FULL
```

Comment:
This logic is repeated in `handleSelect`. What do you think about create a function that calculate the right `QuantityEnum`?